### ISSUE TITLE
Builder gets executed even if buildWhen is false

### ISSUE THREAD

**Issue by mexanichp** [felangel/bloc#1914]
Hi, I'm trying to use BLoC as a state management system for my project but not really understand the build when parameter.
For example, I have the following code:
```dart
BlocConsumer<ExampleBloc, ExampleState>(
  listenWhen: (previous, current) => false,
  listener: (context, state) {
    log('listen fired');
  },
  buildWhen: (previous, current) => false,
  builder: (context, state) {
    log('build fired');
    if (state is ExampleInitial) return Text('Initial');
    if (state is ExampleLoading) return Text('Loading');
    if (state is ExampleSuccess) return Text('Success');
    return Text('No data');
  },
)
```
Once the state has changed the listener is not going to be executed because of `listenWhen` flag, however, the builder is executed regardless of what's inside `buildWhen` flag. What changed is the only `setState` is not executed i.e. `Text` widget is not re-rendered but the builder function is still executed each time. On the other side `listener` callback's not being executed if false. Is it expected?

**Comment by alexisbg** [comment]
I am facing the same issue with BlocConsumer buildWhen/builder since the project has been updated to flutter_bloc 6.1.0.

But BlocBuilder buildWhen/builder seem to work as expected.

**Comment by mexanichp** [comment]
Thanks for posting that @alexisbg I tried to downgrade to the 6.0.0 version and the builder function doesn't get executed. 
It sounds like a bug at this point. I pushed the minimum reproducible version here
[6.1.0]( https://github.com/mexanichp/flutter-bloc-research)
[6.0.0](https://github.com/mexanichp/flutter-bloc-research/tree/1c7ea75539950f00c0bf25d550aa1bf83bd83028)

**Comment by alexisbg** [comment]
There is a simple workaround If your project does not require too many BlocConsumer refactoring: you can replace a BlocConsumer by a BlocBuilder nested into a BlocListener. And it's working with flutter_bloc 6.1.0 too.

**Comment by felangel** [comment]
Thanks for pointing this out! It’s definitely a bug and I’ll have a fix shortly 👍

**Comment by felangel** [comment]
Fixed as part of #1916 and released in `flutter_bloc v6.1.1` 🎉
